In [170]:
import sqlite3
import pandas as pd


## Imports the data and explores it
Steps will be:
- Import the data
- Check out the data
- Check the info
- Describe the data
- Check the shape

# RESET DATA VIEW

In [171]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

#TO SEE EVERYTHING

In [172]:
pd.reset_option('all')
#RUN THIS WHEN DONE SEEING EVERYTHING PLS


: boolean
    use_inf_as_null had been deprecated and will be removed in a future
    version. Use `use_inf_as_na` instead.



F:\Anaconda\envs\learn-env\lib\site-packages\pandas\_config\config.py:622: FutureWarning: 
: boolean
    use_inf_as_null had been deprecated and will be removed in a future
    version. Use `use_inf_as_na` instead.

  warnings.warn(d.msg, FutureWarning)


# IMPORTS

In [174]:
# Imports the data
# Imports the data
bom_raw = pd.read_csv("../data/bom.movie_gross.csv.gz")
# imdb_raw = pd.read_sql("../data/im.db.zip") #not working
rt_movie_info_raw = pd.read_csv("../data/rt.movie_info.tsv.gz", delimiter="\t")
rt_reviews_raw = pd.read_csv("../data/rt.reviews.tsv.gz", delimiter="\t", encoding='cp1252')
tmdb_movies_raw = pd.read_csv("../data/tmdb.movies.csv.gz")
tn_movie_budgets_raw = pd.read_csv("../data/tn.movie_budgets.csv.gz")

In [175]:
# imdb_raw and rt_review_data_raw initially don't load due to an encoding error. Here we detect the encoding type for each before inserting it into the codeblock above
with open("../data/rt.reviews.tsv.gz") as rt_reviews:
    print(rt_reviews)
with open("../data/im.db.zip") as imdbraw:
    print(imdbraw)

<_io.TextIOWrapper name='../data/rt.reviews.tsv.gz' mode='r' encoding='cp1252'>
<_io.TextIOWrapper name='../data/im.db.zip' mode='r' encoding='cp1252'>


# Begin Cleaning RT_Reviews

### BOM info

In [176]:
# Check out the data
bom_raw.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [33]:
bom_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


### Rotten Tomatoes Info and Reviews info

In [177]:
rt_movie_info_raw.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [34]:
rt_movie_info_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [178]:
rt_reviews_raw.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [35]:
rt_reviews_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


### The Movies Database info

In [179]:
tmdb_movies_raw.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [36]:
tmdb_movies_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


### The Numbers Database info

In [180]:
tn_movie_budgets_raw.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [37]:
tn_movie_budgets_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


# Begin Cleaning RT Reviews

### Things I notice:
* date needs to be a date dtype
* what is top critic?
* value counts of fresh column
* maybe change rating to a different dtype/format
* skim review for key words?
* I wonder how many ID's there are (1,135)

In [181]:
#This is here to check cleaning work done instead of re-writing repeatedly
rt_reviews_raw.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [182]:
rt_reviews_raw['date'] = pd.to_datetime(rt_reviews_raw['date'])

In [183]:
#Making a df for top critics
top_critics = rt_reviews_raw.loc[rt_reviews_raw['top_critic'] == 1]

In [184]:
top_critics.critic.value_counts()

Roger Ebert           461
James Berardinelli    348
Owen Gleiberman       210
Peter Travers         193
Mick LaSalle          166
                     ... 
Esther Zuckerman        1
Katherine Vu            1
Karen Heller            1
Sandy Cohen             1
Eric Brace              1
Name: critic, Length: 855, dtype: int64

There are 855 "top critics". I'm not sure this will be very helpful.

In [185]:
# making new column for the new ratings
rt_reviews_raw['new_rating'] = rt_reviews_raw['rating']

In [186]:
#removing + and - on letters
rt_reviews_raw.new_rating = rt_reviews_raw.new_rating.str.strip().str.replace('+', '').str.replace('-', '')

In [187]:
#replacing letters with consistent x/y format
rt_reviews_raw.new_rating = rt_reviews_raw.new_rating.str.replace("A", "5/5").str.replace("B", "4/5").str.replace("C", "3/5").str.replace("D", "2/5").str.replace("F", "1/5")

In [188]:
temp = rt_reviews_raw['new_rating'].str.split('/', expand=True)
temp[1].fillna(10, inplace=True)

In [189]:
#changing x/y to decimal numbers
temp[0].fillna(0, inplace=True)
temp[0] = temp[0].str.replace('N', '0')
temp[0] = temp[0].str.replace('R', '0')
temp[0] = temp[0].str.replace('T', '0')
temp[0] = temp[0].str.replace(' ', '.')
temp[0] = temp[0].astype(float)
temp[1] = temp[1].astype(float)
temp[2] = temp[0]/temp[1]
rt_reviews_raw['new_rating'] = temp[2]

In [190]:
rt_reviews_raw.new_rating.value_counts()

0.800000    7147
0.600000    6832
0.500000    4127
0.400000    4072
0.750000    3664
            ... 
1.550000       1
0.960000       1
0.433333       1
1.250000       1
0.310000       1
Name: new_rating, Length: 85, dtype: int64

? D: there are numbers greater than 1

In [191]:
count_table = rt_reviews_raw.new_rating.value_counts()

wrong_rating = count_table.index > 1

wrong_rating.sum()

5

In [192]:
count_table.loc[wrong_rating]

1.05    9
1.10    4
1.50    2
1.55    1
1.25    1
Name: new_rating, dtype: int64

In [193]:
#creating new df cleaned for naming
rt_reviews_cleaned = rt_reviews_raw.loc[rt_reviews_raw['new_rating'] <= 1]

In [194]:
rt_reviews_cleaned.to_csv(' .csv', index=True)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 40898 entries, 0 to 54431
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   id          40898 non-null  int64         
 1   review      35365 non-null  object        
 2   rating      40898 non-null  object        
 3   fresh       40898 non-null  object        
 4   critic      38919 non-null  object        
 5   top_critic  40898 non-null  int64         
 6   publisher   40671 non-null  object        
 7   date        40898 non-null  datetime64[ns]
 8   new_rating  40898 non-null  float64       
dtypes: datetime64[ns](1), float64(1), int64(2), object(5)
memory usage: 3.1+ MB


Rating col is filled in~!

In [195]:
rt_ratings = rt_reviews_cleaned.groupby(['id'], as_index=False)['new_rating'].mean()
rt_ratings
#checking rotten tomatoes rating grouped by id mean

id  new_rating
0        3    0.628097
1        5    0.685000
2        6    0.592683
3        8    0.703750
4       10    0.579918
...    ...         ...
1110  1996    0.636304
1111  1997    0.498913
1112  1998    0.600000
1113  1999    0.588710
1114  2000    0.560833

[1115 rows x 2 columns]

# Rotten Tomatoes Info Cleaning

In [196]:
#Fix Runtime to only numbers
rt_movie_info_raw['runtime'] = rt_movie_info_raw['runtime'].str.replace(' minutes', '')
rt_movie_info_raw['runtime'] = rt_movie_info_raw['runtime'].fillna(0)
rt_movie_info_raw['runtime'] = rt_movie_info_raw['runtime'].astype('float')

In [197]:
# change name of 'rating' column to 'rated'
rt_movie_info_raw = rt_movie_info_raw.rename(columns={'rating': 'rated'})

In [198]:
# creating new df called clean for consistency
rt_movie_info_cleaned = rt_movie_info_raw

In [199]:
#merging ratings mean into info table
rot_tom = rt_ratings.merge(rt_movie_info_cleaned,
                                   on = 'id')

In [ ]:
 .to_csv(' .csv', index=True)

# The Numbers Database Cleaning

In [200]:
rot_tom.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1115 entries, 0 to 1114
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            1115 non-null   int64  
 1   new_rating    1115 non-null   float64
 2   synopsis      1099 non-null   object 
 3   rated         1114 non-null   object 
 4   genre         1114 non-null   object 
 5   director      997 non-null    object 
 6   writer        879 non-null    object 
 7   theater_date  986 non-null    object 
 8   dvd_date      986 non-null    object 
 9   currency      299 non-null    object 
 10  box_office    299 non-null    object 
 11  runtime       1115 non-null   float64
 12  studio        412 non-null    object 
dtypes: float64(2), int64(1), object(10)
memory usage: 122.0+ KB


In [201]:
rot_tom.groupby('id').first()
#checking the group by


new_rating                                           synopsis  rated  \
id                                                                           
3       0.628097  New York City, not-too-distant-future: Eric Pa...      R   
5       0.685000  Illeana Douglas delivers a superb performance ...      R   
6       0.592683  Michael Douglas runs afoul of a treacherous su...      R   
8       0.703750  The year is 1942. As the Allies unite overseas...     PG   
10      0.579918  Some cast and crew from NBC's highly acclaimed...  PG-13   
...          ...                                                ...    ...   
1996    0.636304  Forget terrorists or hijackers -- there's a ha...      R   
1997    0.498913  The popular Saturday Night Live sketch was exp...     PG   
1998    0.600000  Based on a novel by Richard Powell, when the l...      G   
1999    0.588710  The Sandlot is a coming-of-age story about a g...     PG   
2000    0.560833  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
id                                                                            
3                     Drama|Science Fiction and Fantasy    David Cronenberg   
5                     Drama|Musical and Performing Arts      Allison Anders   
6                            Drama|Mystery and Suspense      Barry Levinson   
8                                 Drama|Kids and Family         Jay Russell   
10                                               Comedy         Jake Kasdan   
...                                                 ...                 ...   
1996   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1997                 Comedy|Science Fiction and Fantasy        Steve Barron   
1998  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1999    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
2000  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
id                                                                     
3                         David Cronenberg|Don DeLillo  Aug 17, 2012   
5                                       Allison Anders  Sep 13, 1996   
6                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
8                                      Gail Gilchriest   Mar 3, 2000   
10                                          Mike White  Jan 11, 2002   
...                                                ...           ...   
1996                                               NaN  Aug 18, 2006   
1997  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1998                                               NaN   Jan 1, 1962   
1999                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
2000                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office  runtime                 studio  
id                                                                       
3      Jan 1, 2013        $     600,000    108.0      Entertainment One  
5     Apr 18, 2000      NaN         NaN    116.0                    NaN  
6     Aug 27, 1997      NaN         NaN    128.0                    NaN  
8     Jul 11, 2000      NaN         NaN     95.0  Warner Bros. Pictures  
10    Jun 18, 2002        $  41,032,915     82.0     Paramount Pictures  
...            ...      ...         ...      ...                    ...  
1996   Jan 2, 2007        $  33,886,034    106.0        New Line Cinema  
1997  Apr 17, 2001      NaN         NaN     88.0      Paramount Vantage  
1998  May 11, 2004      NaN         NaN    111.0                    NaN  
1999  Jan 29, 2002      NaN         NaN    101.0                    NaN  
2000  Feb 11, 2003      NaN         NaN     94.0      Columbia Pictures  

[1115 rows x 12 columns]

### Things I notice:
* there are no nulls!
* release_date needs to be a date type
* Are there duplicate movie names?
* The production budget, domestic_gross, and worldwide_gross should be converted to integers

In [202]:
#checking work
tn_movie_budgets_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [203]:
tn_movie_budgets_raw.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [204]:
# change release_date to datetime
tn_movie_budgets_raw['release_date'] = pd.to_datetime(tn_movie_budgets_raw['release_date'])

In [205]:
# taking out $ and , in numbers
tn_movie_budgets_raw['production_budget'] = tn_movie_budgets_raw['production_budget'].str.replace('$', '').str.replace(',', '')
tn_movie_budgets_raw['domestic_gross'] = tn_movie_budgets_raw['domestic_gross'].str.replace('$', '').str.replace(',', '')
tn_movie_budgets_raw['worldwide_gross'] = tn_movie_budgets_raw['worldwide_gross'].str.replace('$', '').str.replace(',', '')

In [206]:
# casting production_budget as integer
tn_movie_budgets_raw['production_budget'] = tn_movie_budgets_raw['production_budget'].astype(float)
tn_movie_budgets_raw['domestic_gross'] = tn_movie_budgets_raw['domestic_gross'].astype(float)
tn_movie_budgets_raw['worldwide_gross'] = tn_movie_budgets_raw['worldwide_gross'].astype(float)

In [207]:
# creating new df called clean for consistency
tn_movie_budgets_cleaned = tn_movie_budgets_raw

# BOM Cleaning

In [208]:
bom_raw.head()
#check cleaning

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [209]:
bom_raw.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

### Checking each null column

In [210]:
bom_raw[bom_raw['domestic_gross'].isna()]

title   studio  domestic_gross  \
230              It's a Wonderful Afterlife      UTV             NaN   
298   Celine: Through the Eyes of the World     Sony             NaN   
302                              White Lion    Scre.             NaN   
306                        Badmaash Company     Yash             NaN   
327                      Aashayein (Wishes)  Relbig.             NaN   
537                                   Force     FoxS             NaN   
713                        Empire of Silver     NeoC             NaN   
871                            Solomon Kane     RTWC             NaN   
928                            The Tall Man    Imag.             NaN   
933                   Keith Lemon: The Film      NaN             NaN   
936                     Lula, Son of Brazil     NYer             NaN   
966                          The Cup (2012)     Myr.             NaN   
1017                              Dark Tide      WHE             NaN   
1079                         The Green Wave       RF             NaN   
1268                             22 Bullets    Cdgm.             NaN   
1308             Matru Ki Bijlee Ka Mandola      FIP             NaN   
1340                      The Snitch Cartel       PI             NaN   
1342           All the Boys Love Mandy Lane     RTWC             NaN   
1368                                6 Souls     RTWC             NaN   
1659                             Jessabelle      LGF             NaN   
1681                              14 Blades     RTWC             NaN   
1685        Jack and the Cuckoo-Clock Heart   Shout!             NaN   
1739                              Lila Lila    Crnth             NaN   
1975         Surprise - Journey To The West       AR             NaN   
2392                    Finding Mr. Right 2       CL             NaN   
2468                                 Solace      LGP             NaN   
2595                                  Viral   W/Dim.             NaN   
2825                       Secret Superstar      NaN             NaN   

     foreign_gross  year  
230        1300000  2010  
298         119000  2010  
302          99600  2010  
306          64400  2010  
327           3800  2010  
537        4800000  2011  
713          19000  2011  
871       19600000  2012  
928        5200000  2012  
933        4000000  2012  
936        3800000  2012  
966        1800000  2012  
1017        432000  2012  
1079         70100  2012  
1268      21300000  2013  
1308       6000000  2013  
1340       2100000  2013  
1342       1900000  2013  
1368        852000  2013  
1659       7000000  2014  
1681       3800000  2014  
1685       3400000  2014  
1739       1100000  2014  
1975      49600000  2015  
2392     114700000  2016  
2468      22400000  2016  
2595        552000  2016  
2825     122000000  2017

In [106]:
bom_raw[bom_raw['studio'].isna()]

title studio  domestic_gross foreign_gross  year
210   Outside the Law (Hors-la-loi)    NaN         96900.0       3300000  2010
555         Fireflies in the Garden    NaN         70600.0       3300000  2011
933           Keith Lemon: The Film    NaN             NaN       4000000  2012
1862                 Plot for Peace    NaN          7100.0           NaN  2014
2825               Secret Superstar    NaN             NaN     122000000  2017

In [107]:
bom_raw[bom_raw['foreign_gross'].isna()]

title      studio  domestic_gross  \
222                                  Flipped          WB       1800000.0   
254   The Polar Express (IMAX re-issue 2010)          WB        673000.0   
267                           Tiny Furniture         IFC        392000.0   
269            Grease (Sing-a-Long re-issue)        Par.        366000.0   
280                          Last Train Home       Zeit.        288000.0   
...                                      ...         ...             ...   
3382                               The Quake       Magn.          6200.0   
3383             Edward II (2018 re-release)          FM          4800.0   
3384                                El Pacto        Sony          2500.0   
3385                                The Swan  Synergetic          2400.0   
3386                       An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
222            NaN  2010  
254            NaN  2010  
267            NaN  2010  
269            NaN  2010  
280            NaN  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[1350 rows x 5 columns]

Studio seems to be pointless, will drop.

Will also fill rest of domestic/foreign gross NULLS with 0

In [211]:
#dropping studio
bom_raw.drop('studio', axis = 1, inplace = True)

In [212]:
#changing nulls to 0 and converting to float
bom_raw['domestic_gross']=bom_raw['domestic_gross'].fillna(0).astype(float)
bom_raw['foreign_gross'] = bom_raw['foreign_gross'].str.replace(',','')
bom_raw['foreign_gross']=bom_raw['foreign_gross'].fillna(0).astype(float)

In [213]:
bom_cleaned = bom_raw
#moving df to cleaned

# Cleaning The Movie Database

In [214]:
tmdb_movies_raw.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [136]:
tmdb_movies_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Unnamed: 0         26517 non-null  int64         
 1   genre_ids          26517 non-null  object        
 2   id                 26517 non-null  int64         
 3   original_language  26517 non-null  object        
 4   original_title     26517 non-null  object        
 5   popularity         26517 non-null  float64       
 6   release_date       26517 non-null  datetime64[ns]
 7   title              26517 non-null  object        
 8   vote_average       26517 non-null  float64       
 9   vote_count         26517 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(3), object(4)
memory usage: 2.0+ MB


In [215]:
#changing release_date to datetime
tmdb_movies_raw['release_date'] = pd.to_datetime(tmdb_movies_raw['release_date'])

In [216]:
#removing unnecessary str chars
tmdb_movies_raw['genre_ids'] = tmdb_movies_raw['genre_ids'].str.replace(' ', '')
tmdb_movies_raw['genre_ids'] = tmdb_movies_raw['genre_ids'].str.replace(']', '')
tmdb_movies_raw['genre_ids'] = tmdb_movies_raw['genre_ids'].str.replace('[', '')

In [217]:
#making a dict to convert id to actual string
tmdb_genre_dict = {
    "28":"Action",
    "12":"Adventure",
    "16":"Animation",
    "35":"Comedy",
    "80":"Crime",
    "99":"Documentary",
    "18":"Drama",
    "10751":"Family",
    "14":"Fantasy",
    "36":"History",
    "27":"Horror",
    "10402":"Music",
    "9648":"Mystery",
    "878":"Science Fiction",
    "10770":"TV Movie",
    "53":"Thriller",
    "10752":"War",
    "37":"Western"
}

In [218]:
#changing ids to genre
for ind in tmdb_movies_raw.index:
    for key, value in tmdb_genre_dict.items():
        tmdb_movies_raw['genre_ids'][ind] = tmdb_movies_raw['genre_ids'][ind].replace(key, value)

<ipython-input-218-dc25c83cecf4>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  tmdb_movies_raw['genre_ids'][ind] = tmdb_movies_raw['genre_ids'][ind].replace(key, value)


In [219]:
tmdb_movies_raw.drop('id', axis = 1, inplace = True)

In [220]:
tmdb_movies_cleaned = tmdb_movies_raw

# IMDB Cleaning

In [221]:
conn = sqlite3.Connection("../data/im.db")

In [222]:
q0 = """
SELECT movie_id, primary_title, start_year, runtime_minutes, genres
FROM movie_basics
;
"""

imdb_raw = pd.read_sql(q0, conn)

In [223]:
imdb_raw

movie_id                                primary_title  start_year  \
0       tt0063540                                    Sunghursh        2013   
1       tt0066787              One Day Before the Rainy Season        2019   
2       tt0069049                   The Other Side of the Wind        2018   
3       tt0069204                              Sabse Bada Sukh        2018   
4       tt0100275                     The Wandering Soap Opera        2017   
...           ...                                          ...         ...   
146139  tt9916538                          Kuambil Lagi Hatiku        2019   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141  tt9916706                              Dankyavar Danka        2013   
146142  tt9916730                                       6 Gunn        2017   
146143  tt9916754               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 5 columns]

In [224]:
imdb_raw['title_merge'] = imdb_raw['primary_title'].apply(str.lower)

imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace(':', '')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace(' ', '')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace(',', '')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace('-', '')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace('.', '')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace('partiii', 'part3')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace('partii', 'part2')
imdb_raw['title_merge'] = imdb_raw['title_merge'].str.replace('parti', 'part1')

In [225]:
imdb_raw.head()

movie_id                    primary_title  start_year  runtime_minutes  \
0  tt0063540                        Sunghursh        2013            175.0   
1  tt0066787  One Day Before the Rainy Season        2019            114.0   
2  tt0069049       The Other Side of the Wind        2018            122.0   
3  tt0069204                  Sabse Bada Sukh        2018              NaN   
4  tt0100275         The Wandering Soap Opera        2017             80.0   

                 genres                 title_merge  
0    Action,Crime,Drama                   sunghursh  
1       Biography,Drama  onedaybeforetherainyseason  
2                 Drama       theothersideofthewind  
3          Comedy,Drama               sabsebadasukh  
4  Comedy,Drama,Fantasy       thewanderingsoapopera

In [234]:
tn_movie_budgets_cleaned.head()

id release_date                                        movie  \
0   1   2009-12-18                                       Avatar   
1   2   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   3   2019-06-07                                 Dark Phoenix   
3   4   2015-05-01                      Avengers: Age of Ultron   
4   5   2017-12-15            Star Wars Ep. VIII: The Last Jedi   

   production_budget  domestic_gross  worldwide_gross  
0        425000000.0     760507625.0     2.776345e+09  
1        410600000.0     241063875.0     1.045664e+09  
2        350000000.0      42762350.0     1.497624e+08  
3        330600000.0     459005868.0     1.403014e+09  
4        317000000.0     620181382.0     1.316722e+09

In [261]:
bom_cleaned.head()

title  domestic_gross  year  \
0                                  Toy Story 3     415000000.0  2010   
1                   Alice in Wonderland (2010)     334200000.0  2010   
2  Harry Potter and the Deathly Hallows Part 1     296000000.0  2010   
3                                    Inception     292600000.0  2010   
4                          Shrek Forever After     238700000.0  2010   

   worldwide_gross  production_budget  
0     1.067000e+09                  0  
1     1.025500e+09                  0  
2     9.603000e+08                  0  
3     8.283000e+08                  0  
4     7.526000e+08                  0

In [251]:
bom_tomerge.head()

release_date                                        movie  \
0          2010                                  Toy Story 3   
1          2010                   Alice in Wonderland (2010)   
2          2010  Harry Potter and the Deathly Hallows Part 1   
3          2010                                    Inception   
4          2010                          Shrek Forever After   

   production_budget  domestic_gross  worldwide_gross  \
0                  0     415000000.0     1.067000e+09   
1                  0     334200000.0     1.025500e+09   
2                  0     296000000.0     9.603000e+08   
3                  0     292600000.0     8.283000e+08   
4                  0     238700000.0     7.526000e+08   

                            title_merge  
0                             toystory3  
1               aliceinwonderland(2010)  
2  harrypotterandthedeathlyhallowspart1  
3                             inception  
4                     shrekforeverafter

In [228]:
bom_tomerge = bom_cleaned
tn_movie_budgets_tomerge = tn_movie_budgets_cleaned

In [230]:
bom_tomerge['worldwide_gross'] = bom_tomerge['domestic_gross'] + bom_tomerge['foreign_gross']

In [235]:
bom_tomerge['production_budget'] = 0

In [231]:
bom_tomerge.drop('foreign_gross', axis = 1, inplace = True)

In [240]:
bom_tomerge = bom_tomerge.loc[:,['year','title','production_budget','domestic_gross','worldwide_gross']]

In [246]:
bom_tomerge = bom_tomerge.rename(columns={'year': 'release_date'})

In [248]:
bom_tomerge = bom_tomerge.rename(columns={'title': 'movie'})

In [250]:
bom_tomerge['title_merge'] = bom_tomerge['movie'].apply(str.lower)

bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace(':', '')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace(' ', '')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace(',', '')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace('-', '')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace('.', '')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace('partiii', 'part3')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace('partii', 'part2')
bom_tomerge['title_merge'] = bom_tomerge['title_merge'].str.replace('parti', 'part1')

In [244]:
tn_movie_budgets_tomerge.drop('id', axis = 1, inplace = True)

In [253]:
tn_movie_budgets_tomerge

release_date                                        movie  \
0      2009-12-18                                       Avatar   
1      2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2      2019-06-07                                 Dark Phoenix   
3      2015-05-01                      Avengers: Age of Ultron   
4      2017-12-15            Star Wars Ep. VIII: The Last Jedi   
...           ...                                          ...   
5777   2018-12-31                                       Red 11   
5778   1999-04-02                                    Following   
5779   2005-07-13                Return to the Land of Wonders   
5780   2015-09-29                         A Plague So Pleasant   
5781   2005-08-05                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  \
0           425000000.0     760507625.0     2.776345e+09   
1           410600000.0     241063875.0     1.045664e+09   
2           350000000.0      42762350.0     1.497624e+08   
3           330600000.0     459005868.0     1.403014e+09   
4           317000000.0     620181382.0     1.316722e+09   
...                 ...             ...              ...   
5777             7000.0             0.0     0.000000e+00   
5778             6000.0         48482.0     2.404950e+05   
5779             5000.0          1338.0     1.338000e+03   
5780             1400.0             0.0     0.000000e+00   
5781             1100.0        181041.0     1.810410e+05   

                               title_merge  
0                                   avatar  
1     piratesofthecaribbeanonstrangertides  
2                              darkphoenix  
3                      avengersageofultron  
4                starwarsepviiithelastjedi  
...                                    ...  
5777                                 red11  
5778                             following  
5779              returntothelandofwonders  
5780                     aplaguesopleasant  
5781                        mydatewithdrew  

[5782 rows x 6 columns]

In [252]:
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['movie'].apply(str.lower)

tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace(':', '')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace(' ', '')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace(',', '')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace('-', '')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace('.', '')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace('partiii', 'part3')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace('partii', 'part2')
tn_movie_budgets_tomerge['title_merge'] = tn_movie_budgets_tomerge['title_merge'].str.replace('parti', 'part1')

In [255]:
merged_financials = pd.concat([bom_tomerge, tn_movie_budgets_tomerge], ignore_index=True)

In [257]:
merged_financials

release_date                                        movie  \
0                    2010                                  Toy Story 3   
1                    2010                   Alice in Wonderland (2010)   
2                    2010  Harry Potter and the Deathly Hallows Part 1   
3                    2010                                    Inception   
4                    2010                          Shrek Forever After   
...                   ...                                          ...   
9164  2018-12-31 00:00:00                                       Red 11   
9165  1999-04-02 00:00:00                                    Following   
9166  2005-07-13 00:00:00                Return to the Land of Wonders   
9167  2015-09-29 00:00:00                         A Plague So Pleasant   
9168  2005-08-05 00:00:00                            My Date With Drew   

      production_budget  domestic_gross  worldwide_gross  \
0                   0.0     415000000.0     1.067000e+09   
1                   0.0     334200000.0     1.025500e+09   
2                   0.0     296000000.0     9.603000e+08   
3                   0.0     292600000.0     8.283000e+08   
4                   0.0     238700000.0     7.526000e+08   
...                 ...             ...              ...   
9164             7000.0             0.0     0.000000e+00   
9165             6000.0         48482.0     2.404950e+05   
9166             5000.0          1338.0     1.338000e+03   
9167             1400.0             0.0     0.000000e+00   
9168             1100.0        181041.0     1.810410e+05   

                               title_merge  
0                                toystory3  
1                  aliceinwonderland(2010)  
2     harrypotterandthedeathlyhallowspart1  
3                                inception  
4                        shrekforeverafter  
...                                    ...  
9164                                 red11  
9165                             following  
9166              returntothelandofwonders  
9167                     aplaguesopleasant  
9168                        mydatewithdrew  

[9169 rows x 6 columns]

In [259]:
df_imdb_w_fin = imdb_raw.merge(
    merged_financials, 
    left_on='primary_title',
    right_on = 'movie',
    how = 'left')

In [260]:
df_imdb_w_fin[df_imdb_w_fin['domestic_gross'].notnull() ]

movie_id  primary_title  start_year  runtime_minutes  \
20      tt0249516     Foodfight!        2012             91.0   
33      tt0293429  Mortal Kombat        2021              NaN   
38      tt0315642          Wazir        2016            103.0   
40      tt0326592  The Overnight        2010             88.0   
48      tt0337692    On the Road        2012            124.0   
...           ...            ...         ...              ...   
147721  tt9889072    The Promise        2017              NaN   
147770  tt9893078        Sublime        2019              NaN   
147809  tt9899880       Columbus        2018             85.0   
147862  tt9906218    Unstoppable        2019             84.0   
147863  tt9906218    Unstoppable        2019             84.0   

                          genres title_merge_x         release_date  \
20       Action,Animation,Comedy    foodfight!  2012-12-31 00:00:00   
33      Action,Adventure,Fantasy  mortalkombat  1995-08-18 00:00:00   
38            Action,Crime,Drama         wazir                 2016   
40                          None  theovernight  2015-06-19 00:00:00   
48       Adventure,Drama,Romance     ontheroad                 2012   
...                          ...           ...                  ...   
147721                     Drama    thepromise  2017-04-21 00:00:00   
147770               Documentary       sublime  2007-03-13 00:00:00   
147809                    Comedy      columbus  2017-08-04 00:00:00   
147862               Documentary   unstoppable                 2010   
147863               Documentary   unstoppable  2010-11-12 00:00:00   

                movie  production_budget  domestic_gross  worldwide_gross  \
20         Foodfight!         45000000.0             0.0          73706.0   
33      Mortal Kombat         20000000.0      70433227.0      122133227.0   
38              Wazir                0.0       1100000.0        1100000.0   
40      The Overnight           200000.0       1109808.0        1165996.0   
48        On the Road                0.0        744000.0        8744000.0   
...               ...                ...             ...              ...   
147721    The Promise         90000000.0       8224288.0       10551417.0   
147770        Sublime          1800000.0             0.0              0.0   
147809       Columbus           700000.0       1017107.0        1110511.0   
147862    Unstoppable                0.0      81600000.0      167800000.0   
147863    Unstoppable         95000000.0      81562942.0      165720921.0   

       title_merge_y  
20        foodfight!  
33      mortalkombat  
38             wazir  
40      theovernight  
48         ontheroad  
...              ...  
147721    thepromise  
147770       sublime  
147809      columbus  
147862   unstoppable  
147863   unstoppable  

[7181 rows x 12 columns]